accounts-db/src/tiered_storage.rs

```diff
@@ -62,11 +63,16 @@ pub struct TieredStorage {
 
 impl Drop for TieredStorage {
     fn drop(&mut self) {
-        if let Err(err) = fs::remove_file(&self.path) {
+        if let Err(_err) = fs::remove_file(&self.path) {
+            // promote this to panic soon.
+            // disabled due to many false positive warnings while running tests.
+            // blocked by rpc's upgrade to jsonrpc v17
+            /*
             panic!(
                 "TieredStorage failed to remove backing storage file '{}': {err}",
                 self.path.display(),
-            );
+            );*/
+            inc_new_counter_info!("append_vec_drop_fail", 1);
```

Comment:
This section was copied from the AppendVec.  The test will fail without this as some tests sometimes create an empty AccountsFile instance without writing it causing nothing to be dropped.

Reply:
The enabled tests here will fail because of this.  But I think skipping no-such-file-or-directory error should address the problem as we previously discussed.

```
thread 'accounts_db::tests::test_accountsdb_scan_account_storage_no_bank_one_slot' panicked at accounts-db/src/tiered_storage.rs:67:13:
TieredStorage failed to remove backing storage file '/var/folders/2_/848_htk13yb0b_r0wg105t400000gn/T/.tmpFVnzMY/run/0.0': No such file or directory (os error 2)
```

Reply:
Fixed in #446 that keeps the panic behavior but exclude io::ErrorKind::NotFound.